lib/parse.js

```diff
@@ -514,6 +519,38 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
             next();
     }
 
+    function peek_next_token_start_or_newline() {
+        var pos = S.pos;
+        for (var in_multiline_comment = false; pos < S.text.length; ) {
+            var ch = get_full_char(S.text, pos);
+            if (NEWLINE_CHARS.has(ch)) {
+                return ch;
+            } else if (in_multiline_comment) {
+                if (ch == "*" && get_full_char(S.text, pos + 1) == "/") {
+                    pos += 2;
+                    in_multiline_comment = false;
+                } else {
+                    pos++;
+                }
+            } else if (!WHITESPACE_CHARS.has(ch)) {
+                if (ch == "/") {
+                    var next_ch = get_full_char(S.text, pos + 1);
+                    if (next_ch == "/") {
+                        return get_full_char(S.text, find_eol());
+                    } else if (next_ch == "*") {
+                        in_multiline_comment = true;
+                        pos += 2;
+                        continue;
+                    }
+                }
+                return ch;
```

Comment:
It's annoying, but this isn't enough :( the whole name is needed so we know if it's "in" or "of".

For example:

`for (using of x)`
`for (using in x)`

Acorn has a recent commit fixing part of this: https://github.com/acornjs/acorn/commit/e37a9c31423db95ee8de97a2e645a702240e2aa8#diff-781c700ea716836ca6843f093b6a7d486849dec5b3c1ab428c6ce286592913dfR107

Reply:
Oh I see, the `identifier_start` check is not sufficient due to `in` and `instanceof`. I have borrowed `ch_starts_binding_identifier` from the Babel parser and improved the lookahead.

Also fixed another parser bug in `read_name`.